### PR TITLE
Generated graphs for all of the <10> stats (history) keys on dashboard

### PR DESCRIPTION
**Pull request by monika-Dianomic**
Generated graphs for all of the 10 keys on dashboard.

Insert and Update scripts are attached here:
[insert.txt](https://github.com/foglamp/foglamp-gui/files/1699026/insert.txt)
[update.txt](https://github.com/foglamp/foglamp-gui/files/1699027/update.txt)

### CODE REVIEW

File: src/app/dashboard/dashboard.component.ts [hunk 1]
```diff
@@ -117,6 +190,71 @@ export class DashboardComponent implements OnInit {
       });
   }
 
+  statsHistoryBufferedGraph(labels, data): void {
+    this.bufferedChart = 'line';
```
Comment (praveen-garg) [inline src/app/dashboard/dashboard.component.ts]: can we get this as chart_type as arg?

File: src/app/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -117,6 +190,71 @@ export class DashboardComponent implements OnInit {
...
+    this.bufferedValues = {
+      labels: labels,
+      datasets: [
+        {
+          label: '',
+          data: data,
+          backgroundColor: 'rgb(100,149,237)',
+        }
+      ]
+    };
```
Comment (praveen-garg) [inline src/app/dashboard/dashboard.component.ts]: can we make this a reusable object?

File: src/app/dashboard/dashboard.component.ts [hunk 3]
```diff
@@ -117,6 +190,71 @@ export class DashboardComponent implements OnInit {
...
+        display: false // fixme: not working
+      },
+      scales: {
+        yAxes: [{
+          ticks: {
+            beginAtZero: true
+          }
+        }]
+      }
+    };
```
Comment (praveen-garg) [inline src/app/dashboard/dashboard.component.ts]: can we make this a reusable object?

File: src/app/dashboard/dashboard.component.html [hunk 4]
```diff
@@ -1,39 +1,18 @@
...
-        <chart [type]="readingChart" [data]="readingValues" [options]="chartOptions"></chart>
-        <br/>
-        <h6>Total since startup:<b> {{statisticsData['READINGS']}} </b></h6>
-      </article>
+    <ng-progress [positionUsing]="'marginLeft'" [minimum]="0.15" [maximum]="1"
+      [speed]="200" [showSpinner]="true" [direction]="'leftToRightIncreased'"
+      [color]="'#1B95E0'" [trickleSpeed]="300" [thick]="true" [ease]="'linear'">
+    </ng-progress>
+
+    <div class="tile" *ngFor="let statistic of statistics">
```
Comment (praveen-garg) [inline src/app/dashboard/dashboard.component.html]: can we filter (use filter from https://lodash.com/docs/4.17.4 itself) this at ViewModel(controller file) level?

File: src/app/dashboard/dashboard.component.ts [hunk 5]
```diff
@@ -9,30 +9,75 @@ import { NgProgress } from 'ngx-progressbar';
   templateUrl: './dashboard.component.html',
   styleUrls: ['./dashboard.component.css']
 })
+
 export class DashboardComponent implements OnInit {
   statisticsData = {};
   statHistoryData = [];
+  statisticsKeys = [];
+  statistics = []
```
Comment (praveen-garg) [inline src/app/dashboard/dashboard.component.ts]: for above 4 objects, please add a docstring..

File: src/app/dashboard/dashboard.component.ts [hunk 6]
```diff
@@ -3,64 +3,148 @@ import { StatisticsService, AlertService } from '../services/index';
 import Utils from '../utils';
 import { MomentDatePipe } from './../pipes/moment-date';
 import { NgProgress } from 'ngx-progressbar';
+import { Key } from 'selenium-webdriver';
```
Comment (mshariq-nerd) [inline src/app/dashboard/dashboard.component.ts]: selenium-webdriver? seems wrong import.

File: src/app/dashboard/dashboard.component.ts [hunk 7]
```diff
@@ -3,64 +3,148 @@ import { StatisticsService, AlertService } from '../services/index';
...
+        this.statistics.forEach(element => {
           o[element.key] = element.value;
         });
         this.statisticsData = o;
         console.log('This is the statisticsData ', this.statisticsData);
+        this.statisticsKeys = Object.keys(this.statisticsData)
+        console.log('keys array', this.statisticsKeys);
+        this.graphsToShow = this.statistics.filter(value => value['key'] == 'READINGS' || value['key'] == 'SENT_1' || value['key'] == 'PURGED')
+        
+        // Rename 'key' to 'itemName' and add a nwe key as named 'id'
```
Comment (mshariq-nerd) [inline src/app/dashboard/dashboard.component.ts]: typo `new` => `nwe`

**Pull request merged**